samza-test/src/test/java/org/apache/samza/test/framework/StreamApplicationIntegrationTestHarness.java

```diff
@@ -295,4 +227,20 @@ public Config getConfig() {
       return this.config;
     }
   }
+
+  @Override
+  protected KafkaProducer createProducer() {
```

Comment:
It seems a bit odd that this test harness needs slightly different serialization/deserialization for createProducer/createConsumer compared to `IntegrationTestHarness`. If it is difficult to make them consistent, could you please put a comment about why you have these overrides?
You could also consider having helpers in `IntegrationTestHarness` to build the configs, just in case someone needs to add something in the future.

Reply:
I added a comment and wired in the serde through configuration.